code-challenges.test.js

```diff
@@ -38,21 +91,78 @@ const filterLetterE = "e"
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+const filterWordsByLetter = (wordsArray, targetLetter) => { //create function taking two parameters
+    return wordsArray.filter(word => word.toLowerCase().includes(targetLetter.toLowerCase()))
+    //use .filter method to create new array
+    //check if lowercase version of word contains lowercase verion of targetLetter.
+    //use .includes to set condition
+  }
+
+
+
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
 
+ //const hand1 = [5, 5, 5, 3, 3]
+// // Expected output: true
+// const hand2 = [5, 5, 3, 3, 4]
+// // Expected output: false
+// const hand3 = [5, 5, 5, 5, 4]
+// // Expected output: false
+// const hand4 = [7, 2, 7, 2, 7]
+// // Expected output: true
+
+  
+  
+  // a) Create a test with expect statements using the variable provided.
+  describe("hasFullHouse", () => {
+    it("checks if a hand has a full house", () => {
+      const hand1 = [5, 5, 5, 3, 3]
+      const hand2 = [5, 5, 3, 3, 4]
+      const hand3 = [5, 5, 5, 5, 4]
+      const hand4 = [7, 2, 7, 2, 7]
+      
+      expect(hasFullHouse(hand1)).toBe(true)
+      expect(hasFullHouse(hand2)).toBe(false)
+      expect(hasFullHouse(hand3)).toBe(false)
+      expect(hasFullHouse(hand4)).toBe(true)
+    })
+  })
+  
 // b) Create the function that makes the test pass.
+const hasFullHouse = (hand) => {
+    const counts = {}
+    for (const num of hand) {
+      if (counts[num]) {
+        counts[num]++
+      } else {
+        counts[num] = 1
+      }
+    }
+  
+    let hasPair = false
+    let hasThreeOfAKind = false
+  
+    for (const num in counts) {
+      if (counts[num] === 2) {
+        hasPair = true
+      }
+      if (counts[num] === 3) {
+        hasThreeOfAKind = true
+      }
+    }
+  
+    return hasPair && hasThreeOfAKind
+  }
+  
 
 // Pseudo code:
```

Comment:
It's awesome that you're diving into this. I think there's a chance to make the explanation of your code even clearer. While researching is fantastic, how about taking a shot at tackling this challenge with the tools you've picked up in class? Even if you don't get to a solution that is a-okay, you're growing. You've got this! 😊